e2e-test/peerforwarder/build.gradle

```diff
@@ -0,0 +1,203 @@
+    targetContainerId createOpenSearchDockerContainer.getContainerId()
+
+    doLast {
+        sleep(90*1000)
+    }
+}
+
+task stopOpenSearchDockerContainer(type: DockerStopContainer) {
+    targetContainerId createOpenSearchDockerContainer.getContainerId()
+
+    doLast {
+        sleep(5*1000)
+    }
+}
+
+/**
+ * End to end test. Spins up OpenSearch and DataPrepper docker containers, then runs the integ test
+ * Stops the docker containers when finished
+ */
+def createEndToEndTest(final String testName, final String includeTestsMatchPattern,
+                       final DockerCreateContainer createDataPrepper1Task, final DockerCreateContainer createDataPrepper2Task) {
+    return tasks.create(testName, Test) {
+        dependsOn build
+        dependsOn startOpenSearchDockerContainer
+        def startDataPrepper1Task = startDataPrepperDockerContainer(createDataPrepper1Task as DockerCreateContainer)
+        def startDataPrepper2Task = startDataPrepperDockerContainer(createDataPrepper2Task as DockerCreateContainer)
+        dependsOn startDataPrepper1Task
+        dependsOn startDataPrepper2Task
+        startDataPrepper1Task.mustRunAfter 'startOpenSearchDockerContainer'
+        startDataPrepper2Task.mustRunAfter 'startOpenSearchDockerContainer'
+        // wait for data-preppers to be ready
+        doFirst {
+            sleep(15*1000)
+        }
+
+        description = 'Runs the raw span integration tests.'
+        group = 'verification'
+        testClassesDirs = sourceSets.integrationTest.output.classesDirs
+        classpath = sourceSets.integrationTest.runtimeClasspath
+
+        filter {
+            includeTestsMatching includeTestsMatchPattern
+        }
+
+        finalizedBy stopOpenSearchDockerContainer
+        def stopDataPrepper1Task = stopDataPrepperDockerContainer(startDataPrepper1Task as DockerStartContainer)
+        def stopDataPrepper2Task = stopDataPrepperDockerContainer(startDataPrepper2Task as DockerStartContainer)
+        finalizedBy stopDataPrepper1Task
+        finalizedBy stopDataPrepper2Task
+        finalizedBy removeDataPrepperDockerContainer(stopDataPrepper1Task as DockerStopContainer)
+        finalizedBy removeDataPrepperDockerContainer(stopDataPrepper2Task as DockerStopContainer)
+        finalizedBy removeDataPrepperNetwork
+    }
+}
+
+// Discovery mode: LOCAL_NODE
+def includeLocalAggregateTestsMatchPattern = "org.opensearch.dataprepper.integration.peerforwarder.EndToEndPeerForwarderTest.testAggregatePipelineWithSingleNodeEndToEnd*"
+
+def createLocalAggregateDataPrepper1Task = createDataPrepperDockerContainer(
+        "localAggregateDataPrepper1", "dataprepper1", 2021, 4900, 4994, "${AGGREGATE_PIPELINE_YAML}", "${DATA_PREPPER_CONFIG_LOCAL_NODE}")
+def createLocalAggregateDataPrepper2Task = createDataPrepperDockerContainer(
+        "localAggregateDataPrepper2", "dataprepper2", 2022, 4901, 4995, "${AGGREGATE_PIPELINE_YAML}", "${DATA_PREPPER_CONFIG_LOCAL_NODE}")
```

Comment:
Why do we need to bind these containers to different ports? They are run in different containers therefore we should not have a port conflict.

Reply:
We are building DockerFile with the following ports. So we need port bindings as both the containers are using same ports.
```
exposePort(21890)
exposePort(4900)
exposePort(4994)
```

Reply:
We probably don't need to expose ports.